specs/jsonschema-validation.md

```diff
@@ -532,19 +532,24 @@ defined by [RFC 2046](#rfc2046).
 
 ### `contentSchema`
 
-If the instance is a string, and if `contentMediaType` is present, this property
-contains a schema which describes the structure of the string.
+If the instance is a string, and if `contentMediaType` is present, this
+property's subschema describes the structure of the string.
 
 This keyword MAY be used with any media type that can be mapped into JSON
 Schema's data model. Specifying such mappings is outside of the scope of this
 specification.
 
-The value of this property MUST be a valid JSON schema. It SHOULD be ignored if
-`contentMediaType` is not present. Accessing the schema through the schema
-location IRI included as part of the annotation will ensure that it is correctly
-processed as a subschema. Using the extracted annotation value directly is only
-safe if the schema is an embedded resource with both `$schema` and an
-absolute IRI `$id`.
+The value of this property MUST be a valid JSON schema. The subschema is
+produced as an annotation.
+
+Since `contentMediaType` is required to provide instruction on how to interpret
+string content, the annotation schema produced by this keyword has no meaning if
+`contentMediaType` is not present.
+
+Accessing the schema through the schema location IRI included as part of the
+annotation will ensure that it is correctly processed as a subschema. Using the
+extracted annotation value directly is only safe if the subschema is an embedded
```

Comment:
"safe" and "correctly processed as a subschema" is vague -- can we say something else? I think this is trying to say that the evaluation behaviour won't be reproducable because the schema is evaluated in isolation, rather than in the context of the surrounding dialect and location identifier (from the containing schema's $schema and $id keywords).  So how about instead saying something like:

> Because this subschema is intended to be processed in isolation, outside of the context of its containing schema, usage of both the `$schema` and `$id` keywords is recommended to ensure predictable and reproducable results.

Reply:
This text was already present.  I just put it in a new paragraph.  I did think it was a bit convoluted.  Happy to update it.

Reply:
I do think it could be okay to use the subschema _in_ its original context, so that should still be addressed.  This is what the original discussion was about.